lib/core/test/unit/Cardano/Pool/MetadataSpec.hs

```diff
@@ -76,26 +75,47 @@ import qualified Data.Text as T
 spec :: Spec
 spec = do
     describe "Cardano.Pool.Registry.getStakePoolMetadata specs" $
-        around (withStaticServer dataDir) $ do
-        it "Loads the example zip" $ \baseUrl -> do
-            tr <- setupLogging
-            res <- getStakePoolMetadata tr (testUrl baseUrl) presentOwners
+        around withFixtures $ do
+
+        it "Loads the example zip" $ \(baseUrl, metadataDir, tr, _) -> do
+            res <- getStakePoolMetadata tr metadataDir (testUrl baseUrl) presentOwners
             res `shouldBe` Right (map Just presentMetas)
 
-        it "Handles a missing pool" $ \baseUrl -> do
-            tr <- setupLogging
-            res <- getStakePoolMetadata tr (testUrl baseUrl) (absentOwner:presentOwners)
+        it "Handles a missing pool" $ \(baseUrl, metadataDir, tr, _) -> do
+            res <- getStakePoolMetadata tr metadataDir (testUrl baseUrl) (absentOwner:presentOwners)
             res `shouldBe` Right (Nothing:map Just presentMetas)
 
-        it "Fails with an unavailable HTTP server" $ \_port -> do
-            tr <- setupLogging
+        it "Fails with an unavailable HTTP server" $ \(_, metadataDir, tr, _) -> do
             let badUrl = "http://localhost:99/master.zip"
-            res <- getStakePoolMetadata tr badUrl presentOwners
+            res <- getStakePoolMetadata tr metadataDir badUrl presentOwners
             case res of
                 Left (FetchErrorDownload msg) ->
                     head (words msg) `shouldBe` "HttpExceptionRequest"
                 _ -> error "expected fetch error but got none"
 
+        it "Caches the zip" $ \(baseUrl, metadataDir, tr, getMsgs) -> do
+            void $ getStakePoolMetadata tr metadataDir (testUrl baseUrl) presentOwners
+
+            res <- getStakePoolMetadata tr metadataDir (testUrl baseUrl) presentOwners
+            res `shouldBe` Right (map Just presentMetas)
+
+            msgs <- map registryLogMsg <$> getMsgs
+            length (filter isMsgDownloadStarted msgs) `shouldBe` 1
+            length (filter isMsgUsingCached msgs) `shouldBe` 1
```

Comment:
**[remark]**

It's kinda cool to be able to test log messages directly.

Reply:
Yes, Neil told me this also. The tests can easily find out what your program is doing.